roachprod: separate certificate distribution from roachprod start

ajwerner opened the pull request:
This PR is derived from https://github.com/cockroachdb/roachprod/pull/206. It adds a new command `roachprod distribute-certs` which generates and distributes certificates to a cluster if none exist. The third commit adds a wipe option to preserve certificates. The fourth commit additionally appends AWS-style DNS names to allow AWS to work in `--secure` mode.

Fixes #38414.

Commit:
roachprod: learn to expand {certs-dir}

Commit:
roachprod: separate certificate distribution from start

Commit:
roachprod: optionally support preserving certs when in wipe

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/38401)
<!-- Reviewable:end -->

Commit:
roachprod: add AWS dns names to generated certificates

Before this change, AWS nodes could not communicate with eachother when using
--secure because they use DNS names which were not part of the certificate.

This logic could probably be more isolated but this patch is expedient.

Release note: None

bobvawter reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 2 of 4 files at r1, 1 of 3 files at r2, 2 of 3 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/38401#-:-LiJO9vA0y8N-jQ4csac:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @kenliu)



<!-- Sent from Reviewable.io -->

kenliu reviewed (approved):
rubber stamping this

ajwerner commented:
bors r=bobvawter
